pkg/controller/gittrack/git_creds_test.go

```diff
@@ -16,13 +16,246 @@ limitations under the License.
 package gittrack
 
 import (
+	"errors"
+
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 	farosv1alpha1 "github.com/pusher/faros/pkg/apis/faros/v1alpha1"
 	gitstore "github.com/pusher/git-store"
+	testutils "github.com/pusher/faros/test/utils"
+	"k8s.io/client-go/util/flowcontrol"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	farosclient "github.com/pusher/faros/pkg/utils/client"
+		"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 var _ = Describe("GitTrack Suite", func() {
+	Describe("fetchGitCredentials", func() {
+		var m testutils.Matcher
+		var r *ReconcileGitTrack
+		var mgr manager.Manager
+		var stop chan struct{}
+
+		var gt farosv1alpha1.GitTrackInterface
+		var secret *corev1.Secret
+		var creds *gitCredentials
+		var credsErr error
+
+		BeforeEach(func() {
+			// Setup the Manager and Controller.  Wrap the Controller Reconcile function so it writes each request to a
+			// channel when it is finished.
+			var err error
+			cfg.RateLimiter = flowcontrol.NewFakeAlwaysRateLimiter()
+			mgr, err = manager.New(cfg, manager.Options{
+				MetricsBindAddress: "0", // Disable serving metrics while testing
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			applier, err := farosclient.NewApplier(cfg, farosclient.Options{})
+			Expect(err).NotTo(HaveOccurred())
+
+			c, err := client.New(mgr.GetConfig(), client.Options{})
+			Expect(err).NotTo(HaveOccurred())
+
+			m = testutils.Matcher{Client: c, FarosClient: applier}
+
+
+			recFn := newReconciler(mgr)
+			r = recFn.(*ReconcileGitTrack)
+
+			stop = StartTestManager(mgr)
+
+			creds = nil
+			credsErr = nil
+			Expect(creds).To(BeNil())
+			Expect(credsErr).To(BeNil())
```

Comment:
Won't this always be true? Or are the tests somehow magically run in-between these assignments and expects? :thinking:

Reply:
Yes these are redundant, I added them just to be doubly sure though, I'll remove them